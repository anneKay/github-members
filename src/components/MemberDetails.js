import React, { useState, useEffect, Suspense, lazy } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { getUser } from '../actions/memberDetailsaction';
import Header from './common/Header';
import Loader from "./common/Loader";
import "../stylesheet/member-details.scss";

const MemberRepo = lazy(() => import("./MemberRepo"));

const MemberDetails = ({location, getUser}) => {
  console.log(location, '')
  const [ loading, setLoading ] = useState(true);
  const [userDetail, setUserDetail] = useState( {});
  const [userDetailError, setUserDetailError] = useState({})

  useEffect(() => {
    async function fetchMemberDetails() {
      if (loading === true) {
        const response = await getUser(location.pathname);
        if(response.message){
          setUserDetailError(response.message);
        } else {
          setUserDetail(response);
        }
        setLoading(false);
      }
    }
    fetchMemberDetails();
  }, [])

  return (
    <>
    <Header />
    <div className="user-container">

    {userDetailError.length > 0 && (
      <div>{userDetailError}</div>
    )}
    {
          Object.keys(userDetail).length > 0 && (
            <>
            <div className="details-container">
            <img alt="user-avatar" className="avatar" src={userDetail.avatar_url}></img>
            <div className="user-details">
            <p>{`Name: ${userDetail.name}` }</p>
            <div className="user-follow-details">
            <div className="follow">Followers: {userDetail.followers}</div>
            <div className="follow">Following: {userDetail.following}</div>
            </div>
            </div>
            </div>
            <Suspense fallback={<Loader />}>
              <MemberRepo url={userDetail.repos_url} />
            </Suspense>
        </>
      )}
      </div>
    </>
  )
}

const mapStateToProps = state => ({
  responseData: state
});

MemberDetails.propTypes = {
  location: PropTypes.object.isRequired,
  getUser: PropTypes.func.isRequired,
};


export default connect(mapStateToProps, { getUser })(MemberDetails);
