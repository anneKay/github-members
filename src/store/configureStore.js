import { applyMiddleware, compose, createStore } from 'redux';
import ReduxThunk from 'redux-thunk';
import createRootReducer from '../reducers';


const configureStore = () => {
  const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    createRootReducer(),
    {},
    composeEnhancer(
      applyMiddleware(
        ReduxThunk
      ),
    ),
  );
  return store;
};

export default configureStore;
