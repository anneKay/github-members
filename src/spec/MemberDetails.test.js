import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import MemberDetails from '../components/MemberDetails';

const mockStore = configureStore([]);
describe('My Connected React-Redux Component', () => {
  let store, component;
  const location = {
    pathname: "",
  }
  beforeEach(() => {
    store = mockStore({
      data: 'sample data',
      error: '',
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <MemberDetails location={location} />
      </Provider>
    );
  })
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});