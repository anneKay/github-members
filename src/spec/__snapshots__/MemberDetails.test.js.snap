// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Array [
  <header
    id="header-container"
  >
    <div
      className="header-items"
    >
      <h4>
        Members
      </h4>
    </div>
  </header>,
  <div
    className="user-container"
  />,
]
`;
